docs/ref/models/lookups.txt

```diff
@@ -94,6 +94,19 @@ following methods:
     ``compiler.compile(expression)`` should be used. The ``compiler.compile()``
     method will take care of calling vendor-specific methods of the expression.
 
+    ``kwargs`` may be used to supply extra context to the ``as_sql``-method
+    for that particular call. This can be useful for Expressions in
+    particular, where ``template`` can be overridden and variables intended
+    for interpolating the ``template``-string can be passed. 3rd-party
+    backends may have different SQL-syntax so they can pass these differences
+    through via ``kwargs``.
+
+    Example::
+
+        self.as_sql(compiler, connection,
+            template="%(function)s(%(arg)s)", function='Print', arg='Hello World'
+        )
```

Comment:
It's not obvious to me where this `self.as_sql()` call lives. Could you try making the example a bit more complete for a newbie reader, similar to https://docs.djangoproject.com/en/dev/ref/models/expressions/#adding-support-in-third-party-database-backends?